### PR TITLE
Handle uncaught error in CompositeFunction::removeTie when tyring to unfix a tied parameter

### LINKED ISSUE #37940: QENS Fitting crash when fixing IsoDiffRot parameter
**Reported by:** Mohamed Aounae

**Description**
There is a crash on the QENS Fitting interface when attempting to `Fix all` parameters from the `EditLocalParameter` dialog. It should be possible to fix this parameter without a crash.

**To Replicate**
1. Open QENS Fitting interface
2. Go to Convolution tab
3. Load a Sample and Resolution file
4. Select Fit Type `IsoDiffRot`
5. Click in the radius parameter box and select `...`
6. Click the `Set` dropdown button and choose `Remove all ties`
7. Click the `Set` dropdown button and choose `Fix all`
8. Click `Ok`
9. Crash!

### PR DESCRIPTION
**Pull request by MohamedAlmaki**
### Description of work
This PR fixes a runtime error that occurs when removing a tie in a composite function. Previously, `CompositeFunction::removeTie` would first try to remove the tie from `CompositeFunction`. If that failed, it would attempt to remove the tie from the associated function's parameter. However, `IFunction::removeTie` calls unfix before finishing, and if the parameter is tied, unfix triggers a runtime error. This caused `CompositeFunction::removeTie` to attempt removing a tied parameter that isn't present in `CompositeFunction`, leading to a crash. This PR resolves the issue by catching the exception on the first attempt.
#### Summary of work

Fixes #37940. 

### To test:
Follow the steps in #37940 and it should work without a crash

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by warunawickramasingha — CHANGES_REQUESTED** [review]
I have followed the steps given in https://github.com/mantidproject/mantid/issues/37940.

But I am still observing a crash at the same step due to a runtime exception raised from [here](https://github.com/mantidproject/mantid/blob/main/Framework/API/src/IFunction.cpp#L184). Please check. Thanks

**Comment by sf1919** [comment]
@MohamedAlmaki  - I think this has been affected by our changes to pre-commit. Have a look at the summary for the approval in the PR to see if you can identify what needs updating in your PR - https://github.com/mantidproject/mantid/pull/37960#pullrequestreview-2298555084

**Comment by warunawickramasingha** [comment]
Thanks for the changes, and no crash observed this time. I'll approve once the CI is green.

**Comment by sf1919** [comment]
@warunawickramasingha CI is green

**Review by warunawickramasingha — APPROVED** [review]
Happy to approve this!

**Pull request merged**
